File: backend/routes/user_film.go
package routes

import (
	"FilmFindr/controller"
	"FilmFindr/middleware"
	"FilmFindr/service"

	"github.com/gin-gonic/gin"
)

func UserFilmRoutes(router *gin.Engine, userFilmController controller.UserFilmController, jwtService service.JWTService) {
	userFilms := router.Group("/api/v1/user-films")
	{
		userFilms.GET("/user/:id", userFilmController.GetUserFilmByUserId)
		userFilms.POST("/", middleware.Authenticate(jwtService), userFilmController.CreateUserFilm)
		userFilms.PATCH("/:id/status", middleware.Authenticate(jwtService), userFilmController.UpdateStatusUserFilm)
	}
}

File: frontend/src/service/userFilm.js
import axios from "axios";
import { BASE_API_URL } from "../utils/constant";

const BASE_URL = `${BASE_API_URL}/user-films`;

export const getUserFilmByUserId = async (id, page) => {
    const response = await axios.get(`${BASE_URL}/user/${id}?page=${page}`, {
        withCredentials: true
    });
    return response.data;
};

export const createUserFilm = async (data) => {
    const response = await axios.post(`${BASE_URL}/`, data, {
        withCredentials: true,
    });
    return response;
};

export const updateUserFilm = async (id, data) => {
    const response = await axios.patch(`${BASE_URL}/${id}/status`, data, {
        withCredentials: true,
    });
    return response;
};
